tests/rptest/tests/pandaproxy_test.py

```diff
@@ -1205,3 +1204,38 @@ def test_consumer(self):
 
         rc_res = c0.remove(auth=(super_username, super_password))
         assert rc_res.status_code == requests.codes.no_content
+
+    @cluster(num_nodes=3)
+    def test_password_change(self):
```

Comment:
This makes sense. I think it's also worth checking that changing the password in the configuration of a live `kafka/client`, doesn't break the connection, and that it can pick up the new credentials when it needs to reconnect. I guess that needs to be a fixture test in `kafka/client/test`. Different PR is fine.

Reply:
OK then I won't close the referenced issue and I'll put these notes in the issue instead.

Reply:
https://github.com/redpanda-data/redpanda/issues/6593#issuecomment-1279011016